### PR TITLE
Update webgpu-vertex-buffers.md

### PR DESCRIPTION
**Pull request by shallow-beach**
- c/e
- little more on typed arrays (particularly views) + MDN ref
  - i could just leave these edits as a disqus comment, as it is a bit out of scope
- emphasize first explicit use of `attributes` on webgpufundamentals

### CODE REVIEW

**Comment by shallow-beach** [comment]
i will try and add all colons in a separate commit of the branch history going forward, for ease of review.

please lmk if that styling (colons before diagrams/code blocks) is not desired and i will stop that part!

File: webgpu/lessons/webgpu-vertex-buffers.md [hunk 1]
```diff
@@ -93,19 +93,21 @@ for `@location(0)`
...
 `arrayStride` to 8.
 
-Next we define an array of attributes. We only have one. `shaderLocation: 0`
+Next we define an array of attributes. We only have one: `shaderLocation: 0`
 corresponds to `location(0)` in our `Vertex` struct. `offset: 0` says the data
 for this attribute starts at byte 0 in the vertex buffer. Finally `format:
 'float32x2'` says we want WebGPU to pull the data out of the buffer as two 32bit
-floating point numbers.
+floating point numbers. Note the `attributes` property is the same entity pictured in the 
+[simplified draw diagram](https://webgpufundamentals.org/webgpu/lessons/resources/webgpu-draw-diagram.svg)
```
Comment (greggman) [inline webgpu/lessons/webgpu-vertex-buffers.md:109]: I would prefer not to link to the svg file. Instead, in the target .md file, add an anchor with an id that starts with 'a-'. For example, in `webgpufundamentals.md`, 

```
<a id="a-webgpu-state-diagram"></a> Maybe a picture will help. ...
```

Then here, you can link to `[simplified draw diagram](webgpu-fundamentals.html#a-webgpu-state-diagram")`

This has a bunch of advantages:

1. Using an id that starts with `a-` means it's unlikely to conflict with auto-generated ids

2. It links to the explanatory text near to the diagram

3. It *should* work across language. Although in this particular case, you'd have to go add the same anchor in the existing fundamentals language .md files.

Comment (shallow-beach) [reply webgpu/lessons/webgpu-vertex-buffers.md]: used the existing anchor

[resolution commit](./154/commits/701f16582d2ae80c9efbf9927932e19531332e2f)

File: webgpu/lessons/webgpu-vertex-buffers.md [hunk 2]
```diff
@@ -744,7 +750,10 @@ We also need to update the per instance data.
   }
 ```
 
-The layout for the per instance data is like this
+Notice we queue'd the *second* view (`F32`) of the typed array,
```
Comment (greggman) [inline webgpu/lessons/webgpu-vertex-buffers.md:753]: I don't understand this comment. We haven't queue'd anything. queue'd isn't word either it's 'queued' and no idea what `F32` and `U8` are here. I think it's fine as it was.

Comment (shallow-beach) [reply webgpu/lessons/webgpu-vertex-buffers.md:753]: `staticVertexValuesF32` and `staticVertexValuesU8`, w reference to the `device.queue.writeBuffer` on line 753 (of new commit). just emphasizing view behavior, but its discussed plenty in [memory layout article](https://webgpufundamentals.org/webgpu/lessons/webgpu-memory-layout.html)

File: webgpu/lessons/webgpu-vertex-buffers.md [hunk 3]
```diff
@@ -672,13 +673,18 @@ function createCircleVertices({
...
 
-Above we make `colorData` which is a `Uint8Array` view of the same
-data as `vertexData`
+Above we make `colorData`, which is a `Uint8Array` view of the same
+data as `vertexData`. We will only `queue` a `writeBuffer` from `vertexData` -
+all the data stored in `colorData` will also be accessible from `vertexData`.
+We could flip the declaration (create `vertexData` *from* `colorData`), 
+and still `writeBuffer` from `vertexData`, with no issue (all `colorData` would still be accessible). 
+See [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Typed_arrays) 
+for a review of typed arrays.
```
[the remark this answers is not in the document]
Comment (shallow-beach) [reply webgpu/lessons/webgpu-vertex-buffers.md:682]: as a beginner, i would appreciate a link to the [memory layout article](https://webgpufundamentals.org/webgpu/lessons/webgpu-memory-layout.html) there. storage buffers article does not introduce 'views' inasmuch as the only typed arrays used in that article have a single view (i was thrown off and had followed in order up to that point, although clearly missed important part of memory layout article).

File: webgpu/lessons/webgpu-vertex-buffers.md (hunk 2 above)
Comment (shallow-beach) [reply webgpu/lessons/webgpu-vertex-buffers.md]: [resolution commit](./154/commits/6b21a7057fa8547610784c0e7ad1214432ff8383)

File: webgpu/lessons/webgpu-vertex-buffers.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (shallow-beach) [reply webgpu/lessons/webgpu-vertex-buffers.md]: [resolution commit](./154/commits/179c5f4297b2f8fcb9c14b3e5884c221a52b3c18)

**Pull request merged**